hybrid/decomposers.py

```diff
@@ -599,3 +698,289 @@ def next(self, state, **runopts):
+    unconditional edge defects. Because defects are rare in available 
+    processors, it can be solved by brute force.
+    
+    '''
+    #A fully yielded subgraph of the target problem
+    proposed_source = proposed_source.subgraph(list(origin_embedding.keys()))
+
+    #Edges not yielded over this subgraph:
+    unyielded_edge_set = unyielded_conditional_edges(emb = origin_embedding,
+                                                     source=proposed_source,
+                                                     target=target)
+    
+    #Remove minimal number of nodes such that edges yielded on subgraph:
+    G = nx.Graph()
+    G.add_edges_from(unyielded_edge_set)
+    for subgraph_nodes in nx.algorithms.components.connected_components(G):
+        cover = all_minimal_covers(G.subgraph(subgraph_nodes).edges)
+        for v in cover[0]:
+            del origin_embedding[v]
+    
+    #Restrict to giant component. 
+    proposed_source = proposed_source.subgraph(list(origin_embedding.keys()))
+    max_cc = max(nx.connected_components(proposed_source), key=len)
+    if len(max_cc) == 0:
+        raise ValueError('The proposed origin embedding contains no variables.' +
+                         + ' This is likely caused by a key error (not using'
+                         + ' geometrically appropriate variable keys.')
+    origin_embedding = {k:v for k,v in origin_embedding.items() if k in max_cc}
+    return origin_embedding
+
+def make_cubic_lattice(dimensions):
+    ''' Returns an open boundary cubic lattice graph as function of lattice
+    dimensions. Helper function for ``make_origin_embeddings`` '''
+    cubic_lattice_graph = nx.Graph()
+    cubic_lattice_graph.add_edges_from([((x, y, z),(x+1,y,z))
+                                    for x in range(dimensions[0]-1)
+                                    for y in range(dimensions[1])
+                                    for z in range(dimensions[2])])
+    cubic_lattice_graph.add_edges_from([((x, y, z),(x,y+1,z))
+                                    for x in range(dimensions[0])
+                                    for y in range(dimensions[1]-1)
+                                    for z in range(dimensions[2])])
+    cubic_lattice_graph.add_edges_from([((x, y, z),(x,y,z+1))
+                                    for x in range(dimensions[0])
+                                    for y in range(dimensions[1])
+                                    for z in range(dimensions[2]-1)])
+    return cubic_lattice_graph
+
+def make_origin_embeddings(qpu_sampler = None, lattice_type = None):
+    """Creates optimal embeddings for a lattice.
+ 
+    The embeddings created are compatible with the topology and shape of a 
+    specified ``qpu_sampler``. 
+
+    Args:
+        qpu_sampler (:class:`dimod.Sampler`, optional):
+            Quantum sampler such as a D-Wave system.
+            By default \ :class:`~dwave.system.samplers.DWaveSampler()` is
```

Comment:
The class needs to be specified as :class:`~dwave.system.samplers.DWaveSampler` to get intersphinx to render a link. Also, you can delete the preceding backslash: "By default, :class:`~dwave.system.samplers.DWaveSampler` is "